client/CPlayerInterface.cpp

```diff
@@ -930,21 +930,22 @@ void CPlayerInterface::battleEnd(const BattleResult *br)
 		isAutoFightOn = false;
 		cb->unregisterBattleInterface(autofightingAI);
 		autofightingAI.reset();
-		adventureInt->quickCombatUnlock();
 
 		if (!battleInt)

```

Comment:
Do we need to call it on else? Consider regular battle and --headless mode when there is no UI at all

Reply:
So I expect that in case of headless it will not even rich this place so this can be ignored. Still before we called unblock always and now only if battleInt not present

Reply:
I see. we call it later as well